UserGifts/SpecialGiveGift.php

```diff
@@ -311,13 +311,8 @@ function displayFormAll() {
 		if ( $gifts ) {
 			$out->setPageTitle( $this->msg( 'g-give-all-title', $this->user_name_to )->parse() );
 
-			$output .= '<div class="back-links">
-				<a href="' . htmlspecialchars( $user->getFullURL() ) . '">' .
-					$this->msg( 'g-back-link', $this->user_name_to )->parse() .
-				'</a>
-			</div>
-			<div class="g-message">' .
-				$this->msg( 'g-give-all', $this->user_name_to )->parse() .
+			$output .= '<div class="g-message">' .
+				$this->msg( 'g-give-all', htmlspecialchars( $user->getFullURL() ), $this->user_name_to )->text() .
```

Comment:
Il faut eviter d'avoir a mettre du html dans une chaine à traduire.

ici, construit le  lien  "<a>..</a>" dans la fonction php, et passe la valeur à this->msg (a la place de $this->user_name_to)
de meme pour le "{{int:g-send-gift}}" : contruit le dans la fonction php, et passe le en parametre